esp/esdllib/README.md

```diff
@@ -0,0 +1,1608 @@
+
+```xml
+<es:source xpath="soap:Body/*/*/Row/Options/StartDate">
+```
+
+This modifies the default root node of the source of all child operations. The `@xpath` attribute speficies the node to make the default source root node.
+
+```xml
+<es:set-value target="Options/StartDate" select="Year"/>
+```
+
+This sets the value of the target node specified by `@target` to the value of the node specified by `@source`. Any previous value of `@target` is replaced by the new value of `@source`. Because this command is a child of the previous `<es:source>` command, the `@select` is relative to `soap:Body/*/*/Row/Options/StartDate`
+
+```xml
+<es:append-to-value target="Options/StartDate" select="Month"/>
+<es:append-to-value target="Options/StartDate" select="Day"/>
+```
+
+These both append the value of the `@select` node to the value of the `@target` node.
+
+## Transaction Flow and Entry Points
+
+Which version reads better?
+
+### Illustrated by Table
+
+The lifecycle of an ESP transaction has the steps below in the _Transaction Step_ column performed in order. Optionally, scripts can be inserted at any of the entry points named in bold type below in the _Script Entry Point_ column. At each entry point the script has the opportunity to transform the source in preparation for passing it on to the next transaction step.
+
+Order | Transaction Step                                 | Script Entry Point
+------|--------------------------------------------------|-------------------
+  1   | Reciept of request from client                   |
+  2   | Authentication and Authorization                 |
+  3   | ESDL Transforms ESP Request to backend request   |
+  4   | | **BackendRequest** : Transform backend request to final form
+  5   | Backend is called, sending backend request       | **Service** : A script is run instead of calling out to the backend
+  6   | Backend response received                        |
+  7   | | **BackendResponse** : Transform backend response
+  8   | ESDL Transforms backend response to initial ESP response |
+  9   | | **ESDL Response** : Transform initial ESP response to final ESP response
+  10  | ESDL Transforms final ESP response to match schema |
+  11  | ESP sends final response to client |
+  12  | | **PreLogging** : Transform Log Request
+  13  | Logging Manager dispatches Log Request
+
+A script is inserted to an entry point by naming its root node after the name of the entry point. A script can replace a backend service call by configuring it as a **Service** entry point.
+
+Additional details about the entry points and script configuration is found in the reference section.
```

Comment:
I think "is" should be "are", as in "Additional details are found...".

Reply:
fixed